napalm/base/helpers.py

```diff
@@ -255,3 +256,41 @@ def as_number(as_number_val):
         return (int(big) << 16) + int(little)
     else:
         return int(as_number_str)
+
+
+def canonical_interface(interface, change, short=False, update_long_to_short=None):
+    '''
+    Function to retun interface canonical name
+    This puposely does not use regex, or first X characters, to ensure
+    there is no chance for false positives. As an example, Po = PortChannel, and
+    PO = POS. With either character or regex, that would produce a false positive.
+    '''
+    if not change:
+        return interface
+
+    def split_on_match(split_interface):
+        '''
+        simple fuction to split on first digit, slash, or space match
+        '''
+        head = split_interface.rstrip(r'/\0123456789 ')
+        tail = split_interface[len(head):].lstrip()
+        return head, tail
+
+    interface_type, interface_number = split_on_match(interface)
+
+    base_interfaces = interface_map['base_interfaces']
+    reverse_mapping = interface_map['reverse_mapping']
+
+    if isinstance(update_long_to_short, dict):
+        base_interfaces.update(update_long_to_short)
+    # check in dict for mapping
+    if base_interfaces.get(interface_type):
+        long_int = base_interfaces.get(interface_type)
+        # return short form if set
+        if short is True:
```

Comment:
Do we want to overload this method with returning the abbreviated name or have a second method? I think it is more logical to have these as two separate methods.

1. Method one returns the canonical name.
2. Method two returns the abbreviated name (from the canonical name).

Reply:
Was waiting on consensus on this